### PR TITLE
Update QUICHE from bac04054b to 8864d08f6

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
Update QUICHE from bac04054b to 8864d08f6
https://github.com/google/quiche/compare/bac04054b..8864d08f6
    
$ git log bac04054b..8864d08f6 --date=short --no-merges --format="%ad %al %s"
    
2022-04-11 diannahu Fix the receipt of GOAWAYs as the server in oghttp2.    
2022-04-11 danzh Automated g4 rollback of changelist 437128224.
2022-04-11 bnc Platformize enforcing header policy in Balsa.
2022-04-11 vasilvv Use relative include paths for OpenSSL and zlib in QUICHE.
2022-04-11 diannahu Handle new/pending streams on receipt of a GOAWAY in oghttp2.
2022-04-11 bnc Fix header guards in QUICHE.
2022-04-11 wub Deprecate --gfe2_reloadable_flag_quic_tls_no_select_cert_if_disconnected.
2022-04-11 fayang Internal change
2022-04-08 rch Make QUICHE pragmas which disable warnings google3-specific.
2022-04-08 haoyuewang Remove connection_option RVCM such that IETF QUIC connection migration is default on.
2022-04-08 wub Change QuicPathChallengeFrame and QuicPathResponseFrame to be inlined quic frame.
2022-04-07 diannahu Update oghttp2 to close streams above a received GOAWAY's last_stream_id.
2022-04-07 renjietang Change default path degrading delay to 4PTOs when PTO is enabled.
2022-04-07 martinduke Minor cleanup of the QUIC-LB library. Take review comments from late in the process and apply them to earlier classes.
2022-04-07 quiche-dev Internal change
    
Signed-off-by: Ryan Hamilton <rch@google.com>

### CODE REVIEW

File: bazel/repositories.bzl [hunk 1]
```diff
@@ -863,13 +863,11 @@ def _com_googlesource_chromium_zlib():
 def _com_github_google_quiche():
     external_http_archive(
         name = "com_github_google_quiche",
-        # Rewrite third_party includes and #pragma clang.
+        # Rewrite third_party includes.
         patch_cmds = ["find . -type f -exec sed -e '\
     /^#include/ s!third_party/boringssl/src/include/!! ;\
     /^#include/ s!third_party/nghttp2/src/lib/includes/!! ;\
     /^#include/ s!third_party/zlib/!! ;\
-    /^#pragma/ s!clang!GCC!; \
```
Comment (danzh2010) [inline bazel/repositories.bzl:871]: Nice!

**Review by danzh2010 — DISMISSED** [review]
LGTM

**Comment by danzh2010** [comment]
HTTP/2 test failed. Maybe related to https://github.com/google/quiche/commit/9af10d5cc822d74d7aea40f7b0f9aeaf786eb728 @diannahu

**Comment by RyanTheOptimist** [comment]
> HTTP/2 test failed. Maybe related to [google/quiche@9af10d5](https://github.com/google/quiche/commit/9af10d5cc822d74d7aea40f7b0f9aeaf786eb728) @diannahu

Thanks! Yes, indeed. @diannahu is working on a fix.

**Comment by diannahu** [comment]
Thanks @danzh2010 and @RyanTheOptimist! Fix just landed in the Googs.

File: bazel/repositories.bzl (hunk 1 above)
Comment (RyanTheOptimist) [reply bazel/repositories.bzl:871]: Thanks! And with another round of fixes, this is completely gone now!

**Comment by RyanTheOptimist** [comment]
> Thanks @danzh2010 and @RyanTheOptimist! Fix just landed in the Googs.

Thanks so much. I've merged this in and it's passing locally. Now to wait for CI :)

**Comment by RyanTheOptimist** [comment]
Ok, this is all green now. @mattklein123, PTAL

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**